grpc/swarm.go

```diff
@@ -163,6 +177,11 @@ func (service *SwarmService) Register(server *Server) {
 // signed identity.MultiAddress of the client it will return its own signed
 // identity.MultiAddress in a PingResponse.
 func (service *SwarmService) Ping(ctx context.Context, request *PingRequest) (*PingResponse, error) {
+	// Check for empty or invalid request fields.
+	if request == nil || request.MultiAddress == nil {
+		return nil, ErrInvalidRequest
```

Comment:
Using something like `ErrInvalidRequest` either needs to be parameterised, or you should create a more specific error value, e.g. `ErrMultiAddressIsNil`. Having only "invalid request" as a message does not tell us what went wrong during debugging.